app/models/crm/option_set.rb

```diff
@@ -41,7 +54,12 @@ def lookup_by_values(category, *values)
     end
 
     def lookup_const(category)
-      const = const_get(category.to_s.pluralize.upcase)
+      const = begin
+                const_get(category.to_s.upcase)
+              rescue NameError
+                const_get(category.to_s.pluralize.upcase)
+              end
```

Comment:
suggest making the intention clearer here. i can understand what it's doing, but i have to read the code pretty closely to do so!

Reply:
Happy to do so! I tried refactoring it a few ways but always came back to this because it's nice and concise, do you have any ideas?

Originally this method only dealt with pluralised constants, which wasn't great when we added a singular constant ending in `advertising`, so you'd have to refer to it as `advertisings` which just felt horrible, so I added the check for singular first, then fall back to plural.